mirascope/core/base/_utils/parse_content_template.py

```diff
@@ -0,0 +1,115 @@
+"""This module provides a function to parse content parts from a prompt template."""
+
+import imghdr
+import re
+import urllib.request
+from typing import Any, List, cast, get_args
+
+from typing_extensions import TypedDict
+
+from ..message_param import BaseMessageParam, ImagePart, TextPart, _Detail, _PartType
+from .format_template import format_template
+
+
+class _Part(TypedDict):
+    template: str
+    type: _PartType
+
+
+class _ImageOptions(TypedDict, total=False):
+    detail: _Detail
+
+
+def _parse_parts(template: str) -> List[_Part]:
+    pattern = r"\{([^:{}]+):([^{}]+)\}"
```

Comment:
short docstring to describe whats happening, especially in regards to the splitting

Reply:
Done.